sdk/go/pulumi/types.go

```diff
@@ -305,6 +305,14 @@ func AnyWithContext(ctx context.Context, v interface{}) AnyOutput {
 	return internal.ToOutputWithOutputType(ctx, anyOutputType, v).(AnyOutput)
 }
 
+// DeferredOutput creates an Output whose value can be later resolved from another Output instance.
+func DeferredOutput[T any](ctx context.Context) (pulumix.Output[T], func(Output)) {
+	output, resolve := internal.DeferredOutput(ctx)
+	result := AnyOutput{internal.GetOutputState(output)}
+	converted, _ := pulumix.ConvertTyped[T](result)
```

Comment:
This conversion fails in the test, so `converted` ends up being an empty output.

I took a different approach here https://github.com/pulumi/pulumi/pull/18135